underscore.comparison.islike.js

```diff
@@ -0,0 +1,70 @@
+/*
+* Tests if an object is like another. This means objects should follow the same
+* structure and arrays should contain the same types.
+*
+* E.g.
+*
+* _.islike(
+*   {name: "James", age: 10, hobbies: ["football", "computer games", "baking"]},
+*   {name: "", age: 0, hobbies: [""]}
+* )
+*/
+(function() {
+  // Establish the root object, `window` in the browser, or `require` it on the server.
+  if (typeof exports === 'object') {
+    _ = module.exports = require('underscore');
+  }
+
+    var islike = function(obj, pattern) {
+        if (typeof pattern === "function") {
+            return obj instanceof pattern;
+        }
+
+        if (typeof obj !== typeof pattern) return false;
+            if (pattern instanceof Array && !(obj instanceof Array)) return false;
+
+                var type = typeof pattern;
+
+                if (type == "object") {
+                    if (pattern instanceof Array) {
+                        if (pattern.length > 0) {
+                            var oTypes = _.uniq(_.map(obj, fTypeof));
+                            var pTypes = _.uniq(_.map(pattern, fTypeof));
+                            if (_.difference(oTypes, pTypes).length) {
```

Comment:
I don't think this is sufficent. You should go key by key and ensure each key matches (recursive like). This also will restrict key order (e.g. `{a: 1, b: 'a'}, {b: '', a: 1}`)